Simplify config save

simgunz opened the pull request:
Remove unuseful operations when saving settings.

`extraintf` needs to be set only once during the enable procedure, not every time we save the settings.

Disabling `extraintf` may have side effects on other LUA interfaces?? Not sure.

simgunz committed:
Simplify config save

ilgaiaz commented:
Need another control when slowsub is deactivate
in function on_click_save add:
local checkinterface = vlc.config.get("lua-intf")
.
.
if checkinterface == "slowsub_looper_intf" then
            vlc.config.set("lua-intf", "")
        end

If another extension change the "lua-intf" option, slowsub doesn't overwrite this parameter

simgunz committed:
Fix avoid disabling other extension interface without asking

When the extension is first enabled it checks if the lua-intf is already set. In that case it ask the user if he wants to enable the extension making any previous extension using the interface not working.

When disabling the extension only unset lua-intf.

ilgaiaz reviewed (commented):
Add control in function activate()

local currentLuaIntf = vlc.config.get("lua-intf")
if cfg.status.first_run or currentLuaIntf ~= "slowsub_looper_intf" then 
..
..

Avoid error if another extension change the Lua interface configuration after the first run.

simgunz committed:
Check if extension is enabled at every start

We cannot rely on the setting entry first_run because another VLC extension can disable the lua interface.

first_run is unuseful at this point so it was removed.

simgunz committed:
Add check to avoid restart dialog loop

If VLC crashes when the extension is enabled the first time, it is possible that the restart flag is set to false but the interface is not activated, so that it will never set the restart flag to true.

At this point every time the user tries to activate the extension the restart dialog will appear and the only solution is to edit the configuration manually.
